tidy3d/plugins/dispersion/fit_fast.py

```diff
@@ -519,6 +522,16 @@ def iterate_fit(self) -> FastFitterData:
         """
 
         model = self.iterate_poles()
+
+        # if any of the poles align with the test frequencies exactly, we reject the new model
+        if any(
+            np.isclose(omega, 1j * pole, rtol=0, atol=OMEGA_POLE_CLOSE_ATOL)
+            or np.isclose(omega, -1j * pole, rtol=0, atol=OMEGA_POLE_CLOSE_ATOL)
+            for omega in self.omega
+            for pole in self.complex_poles
+        ):
+            return self
```

Comment:
do we want to warn or anything here?

Reply:
thanks. @weiliangjin2021 , do you think we should warn here? I actually think the resulting model in these cases is as expected / as desired, so no need to warn the user. The fix is just preventing the fitter from trying to further improve a model which is already quite good, when the improvement would limit in this error. But I'm not sure exactly about your use case.

Reply:
I could also use fp_eps instead of 1e-10. Might be more robust that way.

Reply:
Yeah, since it's accurate enough, I think no need to warn.